analyzer/src/test/kotlin/managers/NpmTest.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.managers
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.beEmpty
+import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+
+import io.mockk.EqMatcher
+import io.mockk.VarargMatcher
+import io.mockk.every
+import io.mockk.mockkConstructor
+import io.mockk.unmockkConstructor
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.Severity
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.PackageManagerConfiguration
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.utils.common.ProcessCapture
+import org.ossreviewtoolkit.utils.test.createTestTempDir
+
+class NpmTest : StringSpec({
+    "The output of the npm command should be parsed correctly" {
```

Comment:
Would it make sense to first (prior to this commit) refactor `Npm` to have a function

```
fun getIssues(npmOuput: String): List<Issue>
```

And then write two test cases:
1. a unit test which checks that the return value is as expected. Works without mocking IMO would make the expected behavior more obvious
2. setup stub / answer for `getIssues()` and verify that `NPM` uses it.

Reply:
I agree that the test currently takes some effort to check whether the output is correctly parsed. OTOH, this could be used as a basis for further tests. For instance, it could be tested here whether correct command line arguments are provided based on the configuration (which is currently probably not tested).

Maybe the approach to mock the execution of external commands could be useful for a certain kind of tests in general. Then some basic functionality could later be extracted to the test-utils project to make this more convenient.